test-data/unit/check-isinstance.test

```diff
@@ -2729,3 +2729,155 @@ if type(x) is not C:
     reveal_type(x)  # N: Revealed type is "__main__.D"
 else:
     reveal_type(x)  # N: Revealed type is "__main__.C"
+
+[case testHasAttrExistingAttribute]
+class C:
+    x: int
```

Comment:
Since we are using `analyze_member_access`, existance of `__getattr__` on a class might affect the result. Should we add a test-case with a class like:

```python
class WithGetAttr:
    def __getattr__(self, arg: str) -> str:
         return arg  # has all attributes

w: WithGetAttr
if hasattr(w, 'some'):
    reveal_type(w.some)  # Any? str?
```

?